packages/blade/src/components/Input/TextInput/__tests__/TextInput.web.test.tsx

```diff
@@ -167,6 +167,18 @@ describe('<TextInput />', () => {
     expect(onFocus).toHaveBeenCalledWith({ name, value: userName });
   });
 
+  it('should handle onClick', async () => {
+    const onClick = jest.fn();
+    const label = 'Enter name';
+    const { getByLabelText } = renderWithTheme(<TextInput label={label} onClick={onClick} />);
+
+    const input = getByLabelText(label);
+
+    await userEvent.click(input);
+    //should be called for onClick
+    expect(onClick).toHaveBeenCalledTimes(1);
```

Comment:
This test seems to fail 🤔 
<img width="563" alt="image" src="https://github.com/razorpay/blade/assets/30949385/98c8f2bc-59c3-4c18-b43f-a8526ba57473">

Reply:
Ohh, will look into it.

Reply:
On debugging the issue, it seemed the input element wasn't receiving the onClick prop. Is it because of event bubbling?
I guess the main DOM input element is in StyledBaseInput , so added the onClick handler to StyledBaseNativeInput. Ran the test, they seem to be passing. I used watch:test:react to test only the changed files. Let me know if it resolves the issue. Thanks. 
<img width="1440" alt="Screenshot 2023-10-24 at 9 48 22 PM" src="https://github.com/razorpay/blade/assets/83279947/73e07539-44d2-4e1f-8f63-3cb21b6fb61c">

Reply:
Sorry about the 2 silly commits , forgot to remove the console.logs and unnecessary code snippets.

Reply:
haha that's ok we don't really care much about commits since they are squashed and merged in master while merging. Feel free to commit as per your comfort :D